astropy/io/fits/tests/test_convenience.py

```diff
@@ -61,3 +60,36 @@ def test_table_to_hdu(self, tmpdir):
         assert isinstance(hdu, fits.BinTableHDU)
         filename = str(tmpdir.join('test_table_to_hdu.fits'))
         hdu.writeto(filename, overwrite=True)
+
+    def test_printdiff(self):
+        """
+        Test that FITSDiff can run the different inputs without crashing.
+        """
+
+        # Testing different string input options
+        assert printdiff(self.data('arange.fits'), self.data('blank.fits')) is None
```

Comment:
This tests that no exception is raised, but it does not test the actual printed output. While one may say that the underlying `FITSDiff` being called is already tested, would still be nice to ensure the convenience function does not change the `FITSDiff` behaviour.

Perhaps use something like http://doc.pytest.org/en/latest/capture.html#accessing-captured-output-from-a-test-function ? There is a  "new in version 3.0" disclaimer on that page, but I can't tell whether it only applies to the text that comes after or not. Note that Astropy currently needs to support pytest 2.x still.